app/controllers/Checkout.scala

```diff
@@ -417,7 +417,11 @@ class Checkout(fBackendFactory: TouchpointBackends, commonActions: CommonActions
       // should a client want not to proxy via this server.
       Ok(Json.obj("Addresses" -> result.Addresses))
     } recover {
-      case e => BadRequest(Json.obj("Message" -> e.getMessage))
+      case error if error.getMessage == "Bad Request" =>
```

Comment:
Will the message equal "Bad Request"? Have you seen this?

Reply:
Another case you may also add which we saw during the incident was the message being equal to: `Your account has expired. Please upgrade your account.`

Reply:
Yeah I tested it locally with a garbage postcode.

Reply:
It's unfortunate here that the abstraction around the http call doesn't model all of the states as this would alleviate having to rely on a String.

Reply:
I'm not keen on the abstraction used here either (a 404 from getAddress shouldn't result in an Exception IMO, since it's perfectly valid for a user to look up a nonexistent postcode).

I'd prefer to handle the responses based on the HTTP status code, but since that has all been abstracted away, this seem like a quick fix.

Reply:
Yeah completely agree -it's outside the scope of this.